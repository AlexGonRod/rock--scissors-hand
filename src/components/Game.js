import React, { useEffect, useState } from "react";
import Icon from './Icon'

const Game = ({ score, myChoice, setScore, computerChoice, counter, setCounter }) => {
  const [playerWin, setPlayerWin] = useState("");


  const Result = () => {
    if (myChoice === 'rock' && computerChoice === 'scissors') {
      setPlayerWin('Win')
      setScore(score + 1)
    } else if (myChoice === 'rock' && computerChoice === 'paper') {
      setPlayerWin('Lose')
      setScore(score - 1)
    } else if (myChoice === 'scissors' && computerChoice === 'paper') {
      setPlayerWin('Win')
      setScore(score + 1)
    } else if (myChoice === 'scissors' && computerChoice === 'rock') {
      setPlayerWin('Lose')
      setScore(score - 1)
    } else if (myChoice === 'paper' && computerChoice === 'rock') {
      setPlayerWin('Win')
      setScore(score + 1)
    } else if (myChoice === 'paper' && computerChoice === 'scissors') {
      setPlayerWin('Lose')
      setScore(score - 1)
    } else {
      setPlayerWin('Draw')
    }
  };

  useEffect(() => {
    const timer =
      counter > 0
        ? setInterval(() => {
          setCounter(counter - 1)
        }, 1000)
        : Result()

    return () => {
      clearInterval(timer)
    }
  }, [counter])

  

  return (
			<div className="game">
				<div className="game__you">
					<span className="text">You</span>
					<Icon
						dataId={`${myChoice}`}
						className={`icon icon-game-${myChoice}`}
						onClick={() => {}}
					/>
				</div>
				{counter === 0 && (
					<>
						<div className="game__play">
							<span className="text">
								You {playerWin}
							</span>
						</div>
						<div className="game__house">
							<span className="text">Bot</span>
							<Icon
								dataId={`${computerChoice}`}
								className={`icon icon-house-${computerChoice}`}
								onClick={() => {}}
							/>
						</div>
					</>
				)}
			</div>
	)
};

export default Game;
