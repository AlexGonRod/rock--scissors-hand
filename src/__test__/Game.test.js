import React from 'react';
import {shallow} from 'enzyme';
import '@testing-library/jest-dom'
import Game from '../components/Game';

describe('<Game />', () => {
    let wrapper = shallow(<Game />);
    beforeEach(() => {
        wrapper = shallow(<Game />);
    });
    test('<Game /> renders', () => {
        expect(wrapper).toMatchSnapshot();
    })
    test('<Game /> renders text', () => {
        expect(wrapper.find('.game__you .text').length).toEqual(1);
        expect(wrapper.find('.game__you .text').text().trim()).toBe('You');
    })
    test('<Game /> renders children when counter != 0', () => {
        const counter = 1;
        const wrapper = shallow(<Game counter={counter}/>)
        expect(wrapper.find('.game__play').length).toEqual(0);
    })
    test('<Game /> renders children when counter = 0', () => {
        const counter = 0;
        const wrapper = shallow(<Game counter={counter}/>)
        expect(wrapper.find('.game__play').length).toEqual(1);
    })
    // This test has to be fixed. Maybe a setTimeout?
    // test('<Game /> renders you win when success', () => {
    //     const counter = 0,
    //         myChoice = 'rock',
    //         computerChoice='scissors',
    //         playerWin = 'Win';

    //     const wrapper = shallow(<Game myChoice={myChoice} computerChoice={computerChoice} counter={counter}/>)
    //     expect(wrapper.find('.game__play .text').text()).toBe('You Win');
    // })
});