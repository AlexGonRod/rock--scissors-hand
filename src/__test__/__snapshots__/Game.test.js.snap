// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Game /> <Game /> renders 1`] = `
<div
  className="game"
>
  <div
    className="game__you"
  >
    <span
      className="text"
    >
      You
    </span>
    <Icon
      className="icon icon-game-undefined"
      dataId="undefined"
      onClick={[Function]}
    />
  </div>
</div>
`;
